### PR TITLE
xtest: pkcs11: Add tests for random number generation

### PR DESCRIPTION
**Pull request by vesajaaskelainen**
Add test case to test calling for C_SeedRandom() and C_GenerateRandom().

Signed-off-by: Vesa Jääskeläinen <vesa.jaaskelainen@vaisala.com>

### CODE REVIEW

File: host/xtest/pkcs11_1000.c [hunk 1]
```diff
@@ -3652,3 +3652,94 @@ static void xtest_pkcs11_test_1015(ADBG_Case_t *c)
...
+
+	Do_ADBG_BeginSubCase(c, "Generate random bytes with zero length buffer");
+
+	memset(buffer, 0xCC, sizeof(buffer));
+
+	rv = C_GenerateRandom(session, buffer, 0);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	/* Verify that while buffer is still 0xCC */
```
Comment (etienne-lms) [inline host/xtest/pkcs11_1000.c:3727]: s/while/the whole/

File: host/xtest/pkcs11_1000.c [hunk 2]
```diff
@@ -3652,3 +3652,94 @@ static void xtest_pkcs11_test_1015(ADBG_Case_t *c)
...
+
+	rv = C_GenerateRandom(session, buffer, 60);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	/* Verify that end of buffer is still 0xCC */
+	for (i = 60; i < sizeof(buffer); i++)
+		if (!ADBG_EXPECT_COMPARE_UNSIGNED(c, buffer[i], ==, 0xCC))
+			break;
+	Do_ADBG_EndSubCase(c, NULL);
```
Comment (etienne-lms) [inline host/xtest/pkcs11_1000.c:3717]: add an empty line above for consistency

**Review by etienne-lms — COMMENTED** [review]
minor comments

File: host/xtest/pkcs11_1000.c (hunk 1 above)
Comment (vesajaaskelainen) [reply host/xtest/pkcs11_1000.c:3727]: OK

File: host/xtest/pkcs11_1000.c (hunk 2 above)
Comment (vesajaaskelainen) [reply host/xtest/pkcs11_1000.c:3717]: OK

**Review by etienne-lms — COMMENTED** [review]
minor comment.

File: host/xtest/pkcs11_1000.c [hunk 3]
```diff
@@ -3652,3 +3652,95 @@ static void xtest_pkcs11_test_1015(ADBG_Case_t *c)
...
+
+	rv = C_SeedRandom(session, buffer, sizeof(buffer));
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	Do_ADBG_EndSubCase(c, NULL);
+
+	Do_ADBG_BeginSubCase(c, "Seed random bytes with zero length buffer");
+
+	memset(buffer, 0xCC, sizeof(buffer));
```
[the remark this answers is not in the document]
Comment (vesajaaskelainen) [reply host/xtest/pkcs11_1000.c:3693]: OK

File: host/xtest/pkcs11_1000.c [hunk 4]
```diff
@@ -3652,3 +3652,95 @@ static void xtest_pkcs11_test_1015(ADBG_Case_t *c)
...
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	Do_ADBG_EndSubCase(c, NULL);
+
+	Do_ADBG_BeginSubCase(c, "Generate random bytes");
+
+	memset(buffer, 0xCC, sizeof(buffer));
+
+	rv = C_GenerateRandom(session, buffer, 60);
```
[the remark this answers is not in the document]
Comment (vesajaaskelainen) [reply host/xtest/pkcs11_1000.c:3709]: OK

File: host/xtest/pkcs11_1000.c [hunk 5]
```diff
@@ -3652,3 +3652,95 @@ static void xtest_pkcs11_test_1015(ADBG_Case_t *c)
...
+		goto out;
+
+	/* Verify that whole buffer is still 0xCC */
+	for (i = 0; i < sizeof(buffer); i++)
+		if (!ADBG_EXPECT_COMPARE_UNSIGNED(c, buffer[i], ==, 0xCC))
+			break;
+
+	rv = C_GenerateRandom(session, NULL, 0);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
```
[the remark this answers is not in the document]
Comment (vesajaaskelainen) [reply host/xtest/pkcs11_1000.c:3735]: OK

**Comment by vesajaaskelainen** [comment]
Updated with the review comments.

**Comment by ruchi393** [comment]
`Reviewed-by: Ruchika Gupta <ruchika.gupta@linaro.org>`

**Comment by etienne-lms** [comment]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`

**Comment by vesajaaskelainen** [comment]
Thanks!

Tags applied.

**Pull request merged**